bats/tests/k8s/helm-install-rancher.bats

```diff
@@ -13,12 +13,95 @@ local_setup() {
     helm repo update
 }
 
+# Get the minimum Kubernetes version the given rancher-latest/rancher helm chart
+# does _not_ support; i.e. the chart supports < [output].
+get_chart_unsupported_version() {
+    output="" # Hack: tell shellcheck about this variable
+    local chart_version=$1
+    run helm show chart rancher-latest/rancher --version "$chart_version"
+    assert_success || return
+    run awk '/^kubeVersion:/ { $1 = ""; print }' <<<"$output"
+    assert_success || return
+    # We only support kubeVersion of form "< x.y.z"
+    assert_output --regexp '^[[:space:]]*<[[:space:]]*[^[:space:]]+$' || return
+    awk '{ print $2 }' <<<"$output"
+}
+
+# Try to determine the best chart version for the current Kubernetes version
+# (as specified in $RD_KUBERNETES_PREV_VERSION).  Saves the resulting version
+# in rancher_chart_version (and should be restored via load_var).
+determine_chart_version() {
+    if [[ -n $RD_RANCHER_IMAGE_TAG ]]; then
+        # If a version is given, check that it's compatible.
+        assert_rancher_chart_compatible "${RD_RANCHER_IMAGE_TAG#v}" || return
+        rancher_chart_version=${RD_RANCHER_IMAGE_TAG#v}
+        save_var rancher_chart_version
+        return
+    fi
+    local default_version
+    default_version=$(rancher_image_tag)
+    default_version=${default_version#v}
+    run --separate-stderr helm search repo --versions rancher-latest/rancher --output json
+    assert_success || return
+    local versions_json=$output
+    run jq_output 'map(.version).[]'
```

Comment:
I prefer to have an empty line on top of most `run` commands, to break up the wall of code into smaller chunks.